bpf-gpl/jump.h

```diff
@@ -23,14 +23,25 @@
 
 // struct cali_tc_state holds state that is passed between the BPF programs.
 // WARNING: must be kept in sync with the definitions in bpf/polprog/pol_prog_builder.go.
+struct cali_tc_state_dst
+{
+	union
+	{
+		__u16 port;
+		struct {
+			__u8 type;
+			__u8 code;
+		} icmp;
+	}u;
+};
 struct cali_tc_state {
 	__be32 ip_src;
 	__be32 ip_dst;
 	__be32 post_nat_ip_dst;
 	__be32 nat_tun_src;
 	__s32 pol_rc;
 	__u16 sport;
-	__u16 dport;
+	struct cali_tc_state_dst dst;
```

Comment:
How about using an anonymous union for this; I think it'd be clearer:
```
...
__u16 sport;
union {
    __u16 dport;
    struct {
        __u8 icmp_type;
        __u8 icmp_code;
    };
};
__u16 post_nat_dport;
...
```

Then you can access it as `state->icmp_type`, `state->dport` etc without needing to name the union.